Personalization for PrestaShop v2.2.0

kryysler opened the pull request:
New features:

* Completely rewritten UI of the module administration. Now works completely inside the iframe with the exception of the shop context switcher.

Code changes:

* Main module file method getContent() has been completely rewritten to handle the new iframe implementation. All POST requests coming from the page are now XHR requests that always get a JSON response.
* The communication between the iframe and the module is done via window.postMessage().
* Module admin page .tpl file was re-done and made much simpler
* Removed almost all translations, as they are now handled completely within the iframe.
* Removed unused code.

Quetzacoalt91 commented on nostotagging.php:
```diff
@@ -201,150 +200,119 @@ public function uninstall()
-		$output = '';
+		// If we are not under a shop context, redirect to the default shop.
+		// All Nosto settings are stored per shop.
+		if (_PS_VERSION_ >= '1.5' && Shop::getContext() !== Shop::CONTEXT_SHOP)
```
You should also check that the multistore has been enabled. If not, you modify only shop.

```php
Shop::isFeatureActive()
```

Quetzacoalt91 commented on nostotagging.php:
```diff
@@ -201,150 +200,119 @@ public function uninstall()
+		{
+			$shop = new Shop(Configuration::get('PS_SHOP_DEFAULT'));
+			if (!Validate::isLoadedObject($shop))
+				throw new PrestaShopException('Shop not found');
```
If the current context is not a shop, you could ask the merchant to change it instead of displaying an error.

Displaying an exception could make him think that something went wrong on its shop.

Quetzacoalt91 commented:
Hi,

I'm sorry but I cannot accept this PR on the current state, there is now too much iFrames in this module. Directly after the module installation, we meet an iframe to be redirected on a page outside PrestaShop. In my opinion, the previous version was better.

We have already talked about these iFrame, and we accepted them in the first version because they were only showing some details about the merchant account on your Website. But now you are using them to manage the whole module configuration.

kryysler commented:
Hi Thomas,

Updated the pull-request with the new admin layout within the module instead of using the iframe for everything.

Also added the following changes:

* Added hook for sending new product data to Nosto right after product has been created
* Added tagging for search terms
* Improved OAuth error messages
* Fixed product price tagging tax display to depend on user active group
* Fixed product availability tagging to also depend on the products visibility in PS 1.5+

Quetzacoalt91 commented:
Hi @kryysler,

Thanks for this changes. The PR has been merged and the new version will be available soon.

Regards

mridang commented:
@Quetzacoalt91 Thanks for the review.
